servicetalk-http-api/src/test/java/io/servicetalk/http/api/BlockingStreamingToStreamingServiceTest.java

```diff
@@ -93,14 +98,26 @@ void setup() {
         mockCtx = new TestHttpServiceContext(DefaultHttpHeadersFactory.INSTANCE, reqRespFactory, mockExecutionCtx);
     }
 
-    @Test
-    void defaultResponseStatusNoPayload() throws Exception {
-        BlockingStreamingHttpService syncService = (ctx, request, response) -> response.sendMetaData().close();
+    @ParameterizedTest(name = "{displayName} [{index}] withEmptyTrailers={0}")
+    @ValueSource(booleans = {false, true})
+    void defaultResponseStatusNoPayload(boolean withEmptyTrailers) throws Exception {
+        BlockingStreamingHttpService syncService = (ctx, request, response) -> {
+            HttpPayloadWriter<Buffer> writer = response.sendMetaData();
+            if (withEmptyTrailers) {
+                writer.trailers();  // accessing trailers before close should preserve trailers in message body
```

Comment:
What's the use-case of empty trailers?

Reply:
Good question. Looking at it with a fresh morning look. 

My original thinking was that if users didn't touch trailers at all, then it's absolutely safe to drop them. But if they did, they could do something like:
```java
HttpHeaders trailers = writer.trailers();
....
trailers.add(name, value);
```
However, if they did that before `close()`, then we will see that. If they added smth after `close()`, it's anyway racy and not guaranteed. 

Another thing is how we treat trailers in all other places:
1. DefaultHttpRequest.toStreamingRequest() or DefaultHttpResponse.toStreamingResponse()

https://github.com/apple/servicetalk/blob/6fc79877e278769c7c1fe758c2e7f04ce9c2d83b/servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequest.java#L277-L283

2. HttpDataSourceTransformations.aggregatePayloadAndTrailers:

https://github.com/apple/servicetalk/blob/6fc79877e278769c7c1fe758c2e7f04ce9c2d83b/servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpDataSourceTransformations.java#L340-L348

Those account for the case when users can grab a reference to trailers, then convert the response and still use that reference to add trailers. Something like:
```java
HttpResponse aggregatedResponse = ...;
HttpHeaders trailers = aggregatedResponse.trailers();
StreamingHttpResponse streamingResponse = aggregatedResponse.toStreamingResponse();
trailers.add(name, value);
return Single.succeeded(streamingResponse);
```
However, now I see that `BlockingStreamingHttpServerResponse` doesn't have conversion methods at all, and this trick won't be possible with `HttpPayloadWriter` anyway bcz we already sent meta-data.

The only real reason to keep it as a null check instead of "null or empty" is to keep it consistent with the above 2 cases.

Reply:
I think keeping the cases consistent makes sense but if there is a way to simplify I would prefer that.

Reply:
+1 for starting in a consistent way and if we feel different, we can update them all together

Reply:
Thank you for the explanation, makes sense to me to keep it consistent for now 👍